src/notifications/settings/settings.ts

```diff
@@ -1,114 +1,129 @@
- 4. Unsubscribes the wallet from all notification topics on Firebase.
+ 4. Updates the notification subscription
  */
-export const removeNotificationSettingsForWallet = (address: string): Promise<void> => {
-  const allSettings = getAllWalletNotificationSettingsFromStorage();
-  const settingsForWallet = allSettings.find((wallet: WalletNotificationSettings) => wallet.address === address);
+export const removeNotificationSettingsForWallet = async (address: string): Promise<void> => {
+  const walletSettings = getAllWalletNotificationSettingsFromStorage();
+  const globalSettings = getAllGlobalNotificationSettingsFromStorage();
+  const settingsForWallet = walletSettings.find((wallet: WalletNotificationSettings) => wallet.address === address);
 
   if (!settingsForWallet) {
-    return Promise.resolve();
+    return;
   }
 
-  const newSettings = allSettings.filter((wallet: WalletNotificationSettings) => wallet.address !== address);
+  const newWalletSettings = walletSettings.filter((wallet: WalletNotificationSettings) => wallet.address !== address);
 
-  return unsubscribeWalletFromAllNotificationTopics(settingsForWallet.type, NOTIFICATIONS_DEFAULT_CHAIN_ID, address).then(() => {
-    notificationSettingsStorage.set(WALLET_TOPICS_STORAGE_KEY, JSON.stringify(newSettings));
-  });
+  publishAndSaveNotificationSettings({ globalSettings, walletSettings: newWalletSettings });
 };
 
 /**
  Function for enabling/disabling all notifications for a group of wallets.
  Also used to batch toggle notifications for a single wallet
  when using the `Allow Notifications` switch in the wallet settings view.
  */
-export function toggleGroupNotifications(
-  wallets: WalletNotificationSettings[],
-  relationship: WalletNotificationRelationshipType,
-  enableNotifications: boolean
-): Promise<void[][] | void[]> {
-  if (enableNotifications) {
-    return Promise.all(
-      // loop through all owned wallets, loop through their topics, subscribe to enabled topics
-      wallets.flatMap((wallet: WalletNotificationSettings) => {
-        const { topics, address } = wallet;
-        // when toggling a whole group, check if notifications
-        // are specifically enabled for this wallet
-        return Object.keys(topics).map((topic: WalletNotificationTopicType) => {
-          if (topics[topic]) {
-            return subscribeWalletToNotificationTopic(relationship, NOTIFICATIONS_DEFAULT_CHAIN_ID, address, topic);
-          } else {
-            return Promise.resolve();
-          }
-        });
-      })
-    );
-  } else {
-    // loop through all owned wallets, unsubscribe from all topics
-    return Promise.all(
-      wallets.map((wallet: WalletNotificationSettings) => {
-        return unsubscribeWalletFromAllNotificationTopics(relationship, NOTIFICATIONS_DEFAULT_CHAIN_ID, wallet.address);
-      })
-    );
-  }
+export async function toggleGroupNotifications(wallets: WalletNotificationSettings[], enableNotifications: boolean): Promise<boolean> {
```

Comment:
Can we change `wallets` to something a little more specific? Unsure what the best name would be here, maybe something like `newWalletNotificationSettings`. Wdyt